### ISSUE TITLE
Client with the currently selected authenticator does not support any combination of challenges that will satisfy the CA

### ISSUE THREAD

**Issue by themarkymark-steem** [certbot/certbot#5405]
## My operating system is (include version):
Ubuntu 16.04

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
OS Package Manager

## I ran this command and it produced this output:
sudo certbot renew --dry-run
or
sudo certbot --nginx -d [hostname]

## Certbot's behavior differed from what I expected because:
¯\_(ツ)_/¯

###### Logs are stored in `/var/log/letsencrypt` by default. Feel free to redact domains, e-mail and IP addresses as you see fit.
2018-01-10 04:20:37,807:INFO:certbot.auth_handler:Performing the following challenges:
2018-01-10 04:20:37,808:CRITICAL:certbot.auth_handler:Client with the currently selected authenticator does not support any combination of challenges that will satisfy the CA.
2018-01-10 04:20:37,810:WARNING:certbot.renewal:Attempting to renew cert (www.coolsite.com) from /etc/letsencrypt/renewal/www.coolsite.com.conf produced an unexpected error: Client with the currently selected authenticator does not support any combination of challenges that will satisfy the CA.. Skipping.
2018-01-10 04:20:37,812:DEBUG:certbot.renewal:Traceback was:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/certbot/renewal.py", line 425, in handle_renewal_request
    main.renew_cert(lineage_config, plugins, renewal_candidate)
  File "/usr/lib/python2.7/dist-packages/certbot/main.py", line 743, in renew_cert
    _get_and_save_cert(le_client, config, lineage=lineage)
  File "/usr/lib/python2.7/dist-packages/certbot/main.py", line 80, in _get_and_save_cert
    renewal.renew_cert(config, domains, le_client, lineage)
  File "/usr/lib/python2.7/dist-packages/certbot/renewal.py", line 297, in renew_cert
    new_certr, new_chain, new_key, _ = le_client.obtain_certificate(domains)
  File "/usr/lib/python2.7/dist-packages/certbot/client.py", line 318, in obtain_certificate
    self.config.allow_subset_of_names)
  File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 68, in get_authorizations
    self._choose_challenges(domains)
  File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 103, in _choose_challenges
    self.authzr[dom].body.combinations)
  File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 374, in gen_challenge_path
    return _find_smart_path(challbs, preferences, combinations)
  File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 411, in _find_smart_path
    _report_no_chall_path()
  File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 442, in _report_no_chall_path
    raise errors.AuthorizationError(msg)
AuthorizationError: Client with the currently selected authenticator does not support any combination of challenges that will satisfy the CA.

## Here is the relevant nginx server block or Apache virtualhost for the domain I am configuring:

server {
    listen 3002;
    server_name www.coolsite.com;

    location / {
        proxy_pass http://coolip:3002;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
     }
}

**Comment by bmw** [comment]
Unfortunately, there is an ongoing incident with Let's Encrypt and they have currently stopped offering the mechanism that Certbot's nginx plugin uses to prove you control that domain. See https://letsencrypt.status.io/pages/incident/55957a99e800baa4470002da/5a55777ed9a9c1024c00b241 for more info.

Hopefully this will be resolved soon but in the meantime if you have to obtain/renew your cert, you have a couple of options. If you're serving files for that domain out of a directory on that server, you can run the following command:
```
sudo certbot --authenticator webroot --webroot-path <path to served directory> --installer nginx -d <domain>
```
If you're not serving files out of a directory on the server, you can temporarily stop your server while you obtain the certificate and restart it after Certbot has obtained the certificate. This would look like:
```
sudo certbot --authenticator standalone --installer nginx -d <domain> --pre-hook "service nginx stop" --post-hook "service nginx stop"
```
These hooks will cause Certbot to automatically stop your server to obtain certificates and then start it again. After running a command like this once, Certbot will remember your settings so `certbot renew` will work in the future.

For other people who find this issue, this affects some of our other plugins as well such as the Apache plugin. All the advice above is identical except you should replace `nginx` with `apache` in the different CLI options.

**Comment by themarkymark-steem** [comment]
When I do the 2nd option, I get an error saying it is unable to connect to domain.  It's trying to connect via port 80, but the site is hosted on a custom port.  Where do I tell it to do the authentication via a custom port?


Thanks for the insanely quick response, I wasn't finding anything online.

**Comment by bmw** [comment]
> When I do the 2nd option, I get an error saying it is unable to connect to domain. It's trying to connect via port 80, but the site is hosted on a custom port. Where do I tell it to do the authentication via a custom port?

If you need Certbot to listen on a different port, you can include `--http-01-port <port>` on the command line, but if I'm understanding you correctly, the problem is that Let's Encrypt server externally is attempting to connect via port 80. Unfortunately, you cannot change this and Let's Encrypt will always attempt to connect to your server via this port.

If you need a cert now and can't (even temporarily) use one of the above options, you can also look at our [manual plugin](https://certbot.eff.org/docs/using.html#manual). I left instructions about how to use this out of my previous response as it can be a little tedious, but it allows you to work around more complex setups. Set `--authenticator manual` on the command line and it will walk you through manually setting up the challenge or alternatively, you can provide scripts for the manual plugin to run to set up and tear down the challenge as described in the link.

> Thanks for the insanely quick response, I wasn't finding anything online.

Thanks for the kind words! Happy to help.

**Comment by themarkymark-steem** [comment]
The sites are listening on custom ports locally (3000 range), but only expose the ssl port (5000 range) to the public.  

I can temporarily change the port, but there are three hosts on here and that's just a big pain to keep doing.  I guess I can add it into a script to modify the host file to expose port 80 then switch it back after.

I'll look to see if the manual way is more or less work.

**Comment by themarkymark-steem** [comment]
Just did it manually, it's not too bad, but obviously can't auto-renew, but the DNS option is quick and painless.

Thanks!

**Comment by bmw** [comment]
Great! In the future if you want to switch back to the old authentication method so `certbot renew` works, just run:
```
sudo certbot --nginx -d <domain> --force-renewal
```
This will cause Certbot to renew your certificate even it's not needed, but it will remember the settings it used to obtain the cert and use them with `certbot renew` going forward.

**Comment by themarkymark-steem** [comment]
Best way to know when the issue is resolved?

**Comment by bmw** [comment]
You can just watch this issue. This is going to be a common problem for people until it is resolved so we can leave it open until then.

**Comment by christopherfowers** [comment]
I apologize for being "that guy" but is there an estimate that anyone knows of for a resolution to this?

**Comment by hola56** [comment]
Hi,

I tried the second option but hit this error:

Starting nginx:                                            [FAILED]

Hook command "service nginx start" returned error code 1
Error output from service:
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] still could not bind()

Problem binding to port 80: Could not bind to IPv4 or IPv6.

Cannot see a new certificate on the server so assuming it didn't work at all. 

Do you have any advice on how to get round this please?

**Comment by iabel** [comment]
Thanks BMW. For those working with apache, this worked for me:

`certbot --authenticator standalone --installer apache -d <yourdomain> --pre-hook "systemctl stop apache2" --post-hook "systemctl start apache2"`

**Comment by taur** [comment]
for debian/ubuntu:
`certbot --authenticator standalone --installer apache -d <yourdomain> --pre-hook "apache2ctl stop" --post-hook "apache2ctl start"`

**Comment by ccoenen** [comment]
For those watching: it hasn't been explicitly said in here, but this appears to be the source: TLS-SNI-01 has been deactivated as authentication scheme because of security issues.
https://community.letsencrypt.org/t/2018-01-09-issue-with-tls-sni-01-and-shared-hosting-infrastructure/49996
https://community.letsencrypt.org/t/2018-01-11-update-regarding-acme-tls-sni-and-shared-hosting-infrastructure/50188

**Comment by artfulrobot** [comment]
[iabel's method](https://github.com/certbot/certbot/issues/5405#issuecomment-356661779) worked for me on Debian 9 (which uses systemd by default).

**Comment by Lonni93** [comment]
Hi everybody,
I didn't understand yet how renew existing certificates obtained with TLS-SNI-01 challenge. Do I have to obtain new certificates with the suggested "webroot" method (that indeed works perfectly for new domains) manually for every domain? What if I have many of them?

Thx to all

**Comment by Rudloff** [comment]
@Lonni93 For domains that expire soon, you can easily renew them like this:
```bash
sudo certbot renew --webroot --cert-name example.com --webroot-path /path/to/webroot
```
For other domains that expire later, you can wait for the next release of certbot that should fix this issue.

**Comment by Yizi** [comment]
When I type the following is asking for the new webroot which is /var/www/html but it doesn't accept this. Any ideas?

`sudo certbot --authenticator webroot --installer nginx`

**Comment by gnulnx** [comment]
I'm using nginx and trying like this:

sudo certbot --authenticator webroot --installer nginx -m dude@dude.com --agree-tos --noninteractive --webroot-path /home/ubuntu/site/ -d oursite.blah.com

certbot fails with a TIMEOUT trying to fetch

http://oursite.blah.com/.well-known/acme-challenge/LCBIafxnKoVQjt91ykJHRGAG_XB7cZs2QHKynSXqII4:

This simply won't work. We only allow hhtps for 1  (required by our payment gateway) and for 2 it's trying access /.well-known/acme...  which doesn't exist on our site and is correctly throwing a proper 404 error...

We provision all our servers with a single click and we have been using certbot in dev, qa, and prod for a long time now so we would REALLY rather stay with what we have because certbot has been awesome for so long now.  I really would rather help find a resolution before coughing up money to purchase a cert, but alas business is business. 

 Is there any way around this yet?   
Is there anything outsides can do to help?

BTW Thanks to all of you for your hard work and I really appreciate you staying on top of the security issues...  Though I wonder how many of us would have been effected?

**Comment by gnulnx** [comment]
Just want to clarify ours is not a reissue problem.  We need to be able to generate new certs for new stacks on the fly.   We never renew certs.  We alway's destroy our entire stack and rebuild as part of our security policy and certbot fit in perfectly with that workflow.

**Comment by gnulnx** [comment]
Sorry for the stream of comments...

We also tried this:

sudo certbot --authenticator standalone --installer nginx --pre-hook "service nginx stop" --post-hook "service nginx start" -d our.domain.blah

And it's the same TIMEOUT error try to reach a non valid url.. which is ODD since the command explicitly stops nginx.. so I'm not sure how it expected to reach any url on our site.

**Comment by schoen** [comment]
John Furr writes:

> certbot fails with a TIMEOUT trying to fetch
> 
> http://oursite.blah.com/.well-known/acme-challenge/LCBIafxnKoVQjt91ykJHRGAG_XB7cZs2QHKynSXqII4:
> 
> This simply won't work. We only allow hhtps for 1  (required by our payment gateway) and for 2 it's trying access /.well-known/acme...  which doesn't exist on our site and is correctly throwing a proper 404 error...

Hi John,

I don't understand how you say you see both of these errors, as
a timeout and a 404 are reported differently on the CA side (from not
allowing inbound port 80 connections at all vs. allowing them but not
finding a particular path).

The /.well-known/acme-challenge path is created by the webroot
authenticator if you've specified --webroot-path correctly and don't have
any web server configuration rules that reroute this path or prevent it
from being served.

You can get better help with this issue on the Let's Encrypt Community
Forum at

https://community.letsencrypt.org/

Please first read

https://community.letsencrypt.org/t/solution-client-with-the-currently-selected-authenticator-does-not-support-any-combination-of-challenges-that-will-satisfy-the-ca/49983

Some people will be able to switch to the HTTP-01 method and others to
the DNS-01 method (based on updating DNS records via an API).

**Comment by gnulnx** [comment]
> Hi John,

> I don't understand how you say you see both of these errors, as
> a timeout and a 404 are reported differently on the CA side (from not
> allowing inbound port 80 connections at all vs. allowing them but not
> finding a particular path).

My apologies.  I mean it would throw a 404 if it made it through.  Port 80 alway's redirects to 443 so if certbot follows the redirects I'd expect our site to return a 404 an not hold the request until TIMEOUT.   But to be clear the  error reported by certbot is a TIMEOUT error.

Basically this

> IMPORTANT NOTES:
> - The following errors were reported by the server:
>
>   Domain: qa.teaquinox.com
>   Type:   connection
>   Detail: Fetching
>   http://qa.teaquinox.com/.well-known/acme-> challenge/tkR_Oa6Nhlh_ppa_ywq5F_2PqcdEdw6gnwrp--QpasQ:
>   Timeout

I will go take a look at the links you sent.. and thanks for a such a fast reply!

**Comment by chongma** [comment]
The certbot is working normally for me now. It must have been fixed. I am on centos 7

**Comment by bibiwan** [comment]
@chongma : I just have tried and it's still not working.

**Comment by Rudloff** [comment]
[TLS-SNI has been re-enabled for most renewals, and for new certificates on some whitelisted hosting providers.](https://community.letsencrypt.org/t/tls-sni-challenges-disabled-for-most-new-issuance/50316)

**Comment by afterbit01** [comment]
I have a few domains with Let’s Encrypt certificate that use tls-sni-01 validation..
Now they fail to renew..

if i do :

certbot --authenticator webroot --webroot-path /var/www/vhosts/www.mydomain.com --installer apache -d www.mydomain.com

the certificate for that domain (and domain alises) is re-generated and the authentication will switch from tls-sni-01 to HTTP-01? correct?

…and next time the command :

certbot – renew

will get the new setting automatically… correct?

**Comment by Lonni93** [comment]
@afterbit01 correct, the new method (webroot) will be used for the next renew

**Comment by afterbit01** [comment]
and the validation switch to HTTP-01? (also for all the aliases associated to the domain? no need to do manually "the switch" for every aliases)

**Comment by juanpasalagua** [comment]
On this issue - I still can't get it to work on DigitalOcean - any one has an ETA when this will be solved?

**Comment by anatolii1712** [comment]
I'm using nginx and trying like this:
certbot --authenticator standalone --installer nginx -d mysite.com --pre-hook "service nginx stop" --post-hook "service nginx start"
certbot fails with a TIMEOUT trying to fetch:

> IMPORTANT NOTES:
>  - The following errors were reported by the server:
> 
>    Domain: mysite.com
>    Type:   connection
>    Detail: Fetching
>    http://mysite.com/.well-known/acme-challenge/:
>    Timeout
> 
>    To fix these errors, please make sure that your domain name was
>    entered correctly and the DNS A/AAAA record(s) for that domain
>    contain(s) the right IP address. Additionally, please check that
>    your computer has a publicly routable IP address and that no
>    firewalls are preventing the server from communicating with the
>    client. If you're using the webroot plugin, you should also verify
>    that you are serving files from the webroot path you provided.

I'm already read the [post ](https://community.letsencrypt.org/t/solution-client-with-the-currently-selected-authenticator-does-not-support-any-combination-of-challenges-that-will-satisfy-the-ca/49983/2)on official forum but no of both methods doesn't work. It's just stacking on:

> Cleaning up challenges

or giving TIMEOUT ERROR

Is there any way around this yet?
Is there anything outsiders can do to help?

PS: Thanks to all of you for your hard work and I really appreciate you staying on top of the security issues...

**Comment by FlippingBinary** [comment]
I'm running Nginx 1.10.3 on an Ubuntu 16.04.3 LTS server and ran into a problem using the HTTP-01 challenge with one of my domains because the `.well-known` directory was not publicly accessible. The least-disruptive way I found to resolve this was to add a directive in the `server` section of my `.conf` file for the domain. The following example tells Nginx where to find the `.well-known` directory and allows anyone to view it:

```
server {
  server_name example.com
  ...
  location ~ /.well-known {
    root /path/to/nginx/webroot
    allow all;
  }
  location / {
    ...
  }
}
```

I tested the configuration with `sudo nginx -t` and after the test passed I enabled it with `sudo service nginx reload`

Note: The root directory in that location section must be the same one that you use in your certbot command:

`sudo certbot --authenticator webroot --webroot-path /path/to/nginx/webroot --installer nginx -d example.com`

I hope this helps at least one other person.

**Comment by schoen** [comment]
Hi @anatolii1712,

This is probably not a Certbot bug but rather relates to something about your network configuration or firewall. Please open a thread on https://community.letsencrypt.org/ and we can try to help diagnose it.

**Comment by nitinkaveriappa** [comment]
I am getting the below error when I tried to get new certificate.

 Domain: nitinkaveriappa.com
   Type:   connection
   Detail: Fetching
   https://nitinkaveriappa.com/.well-known/acme-challenge/L2GUaZEnr9AFszlcTMmSGBez9CDvQoddlLhRLqho_GU:
   Timeout

Is there a solution to this problem?

**Comment by itsmehemant123** [comment]
Any update on the new build of certbot? I have a mix of virtual hosts - some with root, and some without. 

`$ sudo certbot --authenticator webroot --installer nginx` 
worked perfectly for the hosts with root, and on rewew dry run, they all used http instead of tls-sni. 

However, for those without, we're asked to run this - 
`$ sudo certbot --authenticator standalone --installer nginx --pre-hook "nginx -s stop" --post-hook "nginx"` which never worked when I simulated a dry run. It would stop the nginx server, but after cleanup, it could never start it back up. It always had a PID error. 

And while trying to fix it, i got too many certificates issued error and got rate limited. 

😢

**Comment by umutyerebakmaz** [comment]
sudo certbot --authenticator standalone --installer nginx -d example.com -d www.example.com —pre-hook "service nginx stop" --post-hook "service nginx start"

**Comment by bmw** [comment]
We just [released Certbot 0.21.0](https://community.letsencrypt.org/t/certbot-0-21-0-release/50725) which automatically works around this problem in the Apache and Nginx plugins by using the HTTP-01 challenge. If you are using Certbot from your OS package manager and version 0.21.0 isn't available, you should encourage the Certbot package maintainers for your OS to upgrade to a newer version and if my workarounds suggested [here](https://github.com/certbot/certbot/issues/5405#issuecomment-356498627) don't work for you, you can use [certbot-auto](https://certbot.eff.org/docs/install.html#certbot-auto) to install a newer version of Certbot. If you are already using certbot-auto, it will automatically upgrade to the new version.

Thanks you all for your patience!

**Comment by si458** [comment]
can you update the PPA or is that not maintained by yourselves? 
ppa:certbot/certbot

**Comment by anatolii1712** [comment]
Thank you for all your work. Today after update all working greate, like before. Update, create and expand my certificates using the HTTP-01 challenge:
./letsencrypt-auto --authenticator webroot --webroot-path /etc/nginx/html --installer nginx -d mysite.com

**Comment by mikedotexe** [comment]
Thanks to everyone for the updates, especially @bmw . We're caught in the middle of this during a site launch. Getting the new certbot version via certbot-auto works with the `--dry-run` flag.
Unfortunately, now we're caught in a rate limit. I waited an hour and still get the error:
`There were too many requests of a given type :: Error creating new authz :: too many currently pending authorizations: see https://letsencrypt.org/docs/rate-limits/`
At that link I read about pending authorizations:

> If you’re hitting this rate limit frequently you should double-check your client code.

I assume the errant client code in this case was certbot version 0.19? 
At the very bottom of that page, there's a technique mentioned to clear the pending authorizations via a JWS-signed POST. This seems like an advanced solution, but if the logs provide me with the necessary payload to POST, is this how to fix the problem? (Not sure I understand all the moving parts here.)

It would be a bummer to wait a week for these to clear, and I'm trying to wrap my head around how to untangle the pending authorizations that seem to be caused by this.

**Comment by runephilosof** [comment]
The easiest way around this is to create a new account.
I did this by:
```
mv /etc/letsencrypt/account{,.old}
certbot-auto certonly --webroot # whatever arguments needed to issue cert.
mv accounts.old/acme-v01.api.letsencrypt.org/directory/* accounts/acme-v01.api.letsencrypt.org/directory/
rm -rf accounts.old
```

This way you should be able to issue a new cert today, and the old ones can renew again in a week or you can use the new account to issue new certs for those that need renewing.

**Comment by lucode** [comment]
I run a Ubuntu 16.4 server with  python-certbot-nginx version 0.19.1 which does not upgrade due to the simple fact, that there is no package in the PPA. https://launchpad.net/%7Ecertbot/+archive/ubuntu/certbot
Any workaround to get the newest version?

**Comment by MCOfficer** [comment]
just download the latest release and run certbot-auto.
more details: https://certbot.eff.org/docs/intro.html#how-to-run-the-client

**Comment by kimsovan79** [comment]
https://www.snapengage.com/viewcase?c=35d0a595-4b5a-4b18-a565-3e8148dcd61e

**Comment by edencorbin** [comment]
@lucode exact same issue / server / version, did you figure anything out?

**Comment by Ender22** [comment]
Can you guys help me understand why on ubuntu 14 it thinks 0.19.0 is the newest version of certbot? I'm rather new to linux and also first time using certbot.

**Comment by MCOfficer** [comment]
@Ender22 the ppa hasn't been updated yet, and we don't know when this will be the case

**Comment by keros** [comment]
I really hope the ppa gets an update in the next days.
I don't want to manual update all my Ubuntu machines

**Comment by pcraston** [comment]
I emailed the PPA maintainers who replied with the following message:

```
This PPA usually lags behind Debian, and we just let 0.20 migrate to testing yesterday after baking for about a week in unstable. (0.20 we gave additional time, since it migrated to python3 from python2.)

I know there’s important stuff in the 0.21 release, so it’s a definite priority for me. As soon as I have it in Debian unstable, I’ll talk to Ondrej about the PPA.

It probably won’t be ready for at least a week though, taking into account migration times, testing, etc.
```

So just need to be a bit patient ;-) If you need the new version urgently, you need to use certbot-auto I'm afraid.

**Comment by thiagodiogo** [comment]
Thanks  @mikedotexe and @bmw on the support!

**Comment by sebastiancarlsson** [comment]
This worked for me on a Digital Ocean droplet with Ubuntu 16.04 and Nginx (and certbot 0.19).

`sudo certbot --authenticator standalone --installer nginx -d <domain.com> -d <www.domain.com> --pre-hook "systemctl stop nginx" --post-hook "systemctl start nginx"`

**Comment by keros** [comment]
The --standalone option don't uses the disabled tls-sni-01 challenge. So it works fine on 14.04, 16.04, ... .
The problem behind this version is apache/nginx needs to be stopped which is not always possible.
But they work on the new ppa version so everything is ok

**Comment by Fire-Brand** [comment]
@sebastiancarlsson, trying your approach (but on AWS), gave me:
```
   Domain: sub.domain.com
   Type:   unauthorized
   Detail: Incorrect validation certificate for tls-sni-01 challenge.
   Requested
   <really long hash>.acme.invalid
   from <server ip>:443. Received 4 certificate(s), first
   certificate had names "*.domain.com"
```
Any ideas why?

**Comment by rcubitto** [comment]
@sebastiancarlsson your solution worked for me, I have the same specs as you do.

**Comment by CampStove** [comment]
when attempting the first option i am asked to input the webroot for my domain, but i am unsure what my webroot is or how to find it. Does anyone know how i can find out what it is?

**Comment by FlippingBinary** [comment]
@CampStove The webroot is the location where certbot can place (or find) a folder called `.well-known` and have it accessible at `https://example.com/.well-known` where `example.com` is the domain name you are attempting to validate. If you don't know where to look, you could try asking your web server for clues. For Nginx, you can find possible locations by executing the command `sudo nginx -T | grep root` in a terminal. For Apache, you could try `sudo apache2ctl -S | grep DocumentRoot`

Hope this helps.

**Comment by ciprianflorescu** [comment]
`Failed authorization procedure. www.domainname.com (http-01): urn:acme:error:connection :: The server could not connect to the client to verify the domain :: DNS problem: NXDOMAIN looking up A for www.domainname.com

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: www.domainname.com
   Type:   connection
   Detail: DNS problem: NXDOMAIN looking up A for
   www.domainname.com

   To fix these errors, please make sure that your domain name was
   entered correctly and the DNS A/AAAA record(s) for that domain
   contain(s) the right IP address. Additionally, please check that
   your computer has a publicly routable IP address and that no
   firewalls are preventing the server from communicating with the
   client. If you're using the webroot plugin, you should also verify
   that you are serving files from the webroot path you provided.`

My domain works fine btw if you load it on http. What's up with this error?

**Comment by FlippingBinary** [comment]
@ciprianflorescu It looks like letsencrypt can't verify your server's IP address. There is a similar problem mentioned in [this stack overflow question](https://stackoverflow.com/questions/48356095/dns-nxdomain-error-command-certbot). Could you verify that your DNS server is authoritative? One place to do that is at [Zonemaster](https://www.zonemaster.net/).

**Comment by ciprianflorescu** [comment]
@FlippingBinary All Zonemaster tests appear to be fine. No issues were raised.

**Comment by FlippingBinary** [comment]
@ciprianflorescu This may be a temporary problem. I've seen your error message referenced several times in other forums, but as long as the domain name is publicly accessible and the DNS server is authoritative, the only solution seems to be time. Maybe try again in six hours?

**Comment by ciprianflorescu** [comment]
Well I tried to do it using the other method:

`$ sudo certbot --authenticator standalone --installer nginx --pre-hook "nginx -s stop" --post-hook "nginx"`

 and I get:

> Hook command "nginx" returned error code 1
> Error output from nginx:
> nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
> nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
> nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
> nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
> nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
> nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
> nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
> nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
> nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
> nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
> nginx: [emerg] still could not bind()

Is there any method to install a cert that currently works?

**Comment by ciprianflorescu** [comment]
Ok, so I managed to completely stop nginx, but now it's starting to complain about A records again, even though I'm 100% sure the A records on this domain are fine. The site works and domain names work, if I start nginx.

>2018-01-29 21:32:02,974:DEBUG:certbot.reporter:Reporting to user: The following errors were reported
 by the server:

> Domain: www.domain.com
> Type:   connection
> Detail: DNS problem: NXDOMAIN looking up A for www.domain.com
> 
> To fix these errors, please make sure that your domain name was entered correctly and the DNS A/AAAA
>  record(s) for that domain contain(s) the right IP address. Additionally, please check that your com
> puter has a publicly routable IP address and that no firewalls are preventing the server from commun
> icating with the client. If you're using the webroot plugin, you should also verify that you are ser
> ving files from the webroot path you provided.
> 2018-01-29 21:32:02,975:INFO:certbot.auth_handler:Cleaning up challenges
> 2018-01-29 21:32:02,980:DEBUG:certbot.plugins.standalone:Stopping server at :::80...
> 2018-01-29 21:32:03,357:DEBUG:certbot.log:Exiting abnormally:
> Traceback (most recent call last):
>   File "/usr/bin/certbot", line 11, in <module>
>     load_entry_point('certbot==0.19.0', 'console_scripts', 'certbot')()
>   File "/usr/lib/python2.7/dist-packages/certbot/main.py", line 861, in main
>     return config.func(config, plugins)
>   File "/usr/lib/python2.7/dist-packages/certbot/main.py", line 786, in certonly
>     lineage = _get_and_save_cert(le_client, config, domains, certname, lineage)
>   File "/usr/lib/python2.7/dist-packages/certbot/main.py", line 85, in _get_and_save_cert
>     lineage = le_client.obtain_and_enroll_certificate(domains, certname)
>   File "/usr/lib/python2.7/dist-packages/certbot/client.py", line 357, in obtain_and_enroll_certific
> ate
>     certr, chain, key, _ = self.obtain_certificate(domains)
>   File "/usr/lib/python2.7/dist-packages/certbot/client.py", line 318, in obtain_certificate
>     self.config.allow_subset_of_names)
>   File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 81, in get_authorizations
>     self._respond(resp, best_effort)
>   File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 138, in _respond
>     self._poll_challenges(chall_update, best_effort)
>   File "/usr/lib/python2.7/dist-packages/certbot/auth_handler.py", line 202, in _poll_challenges
>     raise errors.FailedChallenges(all_failed_achalls)
> FailedChallenges: Failed authorization procedure. www.domain.com (http-01): urn:acme:error:connection :: The server could not connect to the client to verify the domain :: DNS problem: NXDOMAIN looking up A for www.domain.com

What I did was `sudo certbot certonly --standalone`

**Comment by falconmick** [comment]
Is there a way I can be notified when the fix is avaiable for PPA so I can change my config around?

**Comment by amjad** [comment]
label's command worked! thank you.

**Comment by dugajean** [comment]
The "next few days" have passed. Anything new on this?

**Comment by artemxgruden** [comment]
if there are problems with webroot method and nginx, check how to https://gist.github.com/cecilemuller/a26737699a7e70a7093d4dc115915de8

it helps me ( i use certbot inside docker nginx container on ubuntu:16.04 )

**Comment by pde** [comment]
@dugajean yes, Certbot 0.21 was [released on the 17th of January](https://github.com/certbot/certbot/blob/master/CHANGELOG.md), and should have fixed this problem for just about everyone.  Some considerations:

* We need to update the Certbot website to change the warning or what it points to :)
* You need to actually run Certbot 0.21+. If your OS doesn't have packages for that, [certbot-auto](https://certbot.eff.org/docs/install.html#certbot-auto) is a good temporary solution.
* The HTTP-01 challenge support in the Apache and Nginx plugins for Certbot 0.21.1 is pretty good, though your webserver needs to be reachable on port 80. If you have problems with it, please file a bug.

**Comment by amjad** [comment]
I still get this error when I follow Digital Ocean’s Lets Encrypt tutorial for Ubuntu Apache.

**Comment by chris001** [comment]
@amjad 
https://www.digitalocean.com/community/tutorials/how-to-secure-apache-with-let-s-encrypt-on-ubuntu-16-04
That tutorial is using the ubuntu ppa for `certbot`.
```
$ certbot --version
certbot 0.19.0
```
The error you get is because the ubuntu maintainer for `ppa:certbot/certbot` hasn't updated it to `certbot 0.21.0` yet, that version contains the fix.
You can workaround by modifying your site to get the cert, either run `certbot` in `standalone` mode, or in `webroot` mode, or download and use `cerbot-auto` which contains the fix.

**Comment by fkrauthan** [comment]
Any news on the PPA update? This is really a big problem not having a working version. After all the PPA is marked as the official way for running certbot on ubuntu.

**Comment by falconmick** [comment]
How about instead of the unhappiness we try to help. Is there anything any of us could contribute to help?

**Comment by chris001** [comment]
@falconmick @fkrauthan 
* Not really any way to contribute.
* See https://github.com/certbot/certbot/issues/5405#issuecomment-359405017
@pcraston emailed the ubuntu ppa maintainer for `cerbot` and they said 13 days ago 0.21.0 would be available in about a week, so they're lagging, you should probably run `certbot-auto` now, until `cerbot` 0.21.0 is available, could be tomorrow, could be a week from now:
https://certbot.eff.org/docs/install.html#certbot-auto

**Comment by jmpinto** [comment]
Hi
I would like to ask if I'm not able to get a Certificate if my domain extension is .com.br?
I have a server with Centos 6 and Freepbx (with Apache) and is under a domain .com.br all the time that I try to get the certificate using SSH I'm not able because it says that has  wrong caracter. 
Also using the FreePBX Gui I do not get the Let's Encrypted Certificate
Someone here could help me? Please
Thank you for your time and attention

**Comment by ciprianflorescu** [comment]
@chris001 
>You can workaround by modifying your site to get the cert, either run certbot in standalone mode, or in webroot mode, or download and use cerbot-auto which contains the fix.

I believe I tried all these methods and I still get the error about DNS issues. Is there any detailed, manual installation guide for certbot 0.21? Like downloading the source from a repository to the server, then using that to install certbot. I am running nginx on Ubuntu Trusty 14.04 and they haven't updated their PPA for certbot.

The certbot-auto method didn't work for me either. All these methods eventually lead to either some errors in installation or fail to recognise the domain name. Even though the domain works.

I may just have to give up on letsencrypt and just get a domain with included ssl and call it a day. This whole chain of weak links between Ubuntu maintainers - letsencrypt and whatnot is making me waste time I could use to be more productive. 

Sadly the web world is full of such weak links. We need another model for the web.

**Comment by falconmick** [comment]
Sounds like sombody isn’t happy with there free software, better get him a refund!

In all seriousness though, this is the risk you take when it doesn’t cost a penny, paid certs means that sombody has an incentive to make things as easy as possible for you, so it’s definitely a good option.

My main reason for let’s encrypt is the 3 month auto renewal system. It’s just a better way of handling certs

**Comment by ciprianflorescu** [comment]
Yeah and this is why free software is unreliable as fuck. And not a viable solution for anything reliable, in fact. This model of free software doesn't work. There's not much accountability if any at all, just waiting on the potential availability of devs to do this of their own will.

No need to be a wise ass about it, though. I'm well aware of the implications of using free stuff.

**Comment by FlippingBinary** [comment]
@ciprianflorescu I would like to point out that you could seek paid support from one of many providers. Regardless, I went ahead and spun up a server running Ubuntu 14.04 to demonstrate running certbot-auto and made a video for you on [YouTube](https://youtu.be/0TYjZ_TYr-s). Watch the video if you want. It shows the process working flawlessly.

**Comment by chris001** [comment]
@ciprianflorescu @falconmick @FlippingBinary @fkrauthan 
The Ubuntu PPA for `certbot` is updated now to 0.21.1 - thanks to certbot ppa maintainers. `sudo apt-get ugrade` and then `cerbot` command should work fine for you now.

@jmpinto 
If you use `certbot-auto` on centos 6, it should download the newest version `certbot` 0.21+ and work fine and it will even install your LE TLS cert into apache2, as long as your specify apache2 on the command line. Watch video of @FlippingBinary https://youtu.be/0TYjZ_TYr-s to learn how to run the process of `certbot-auto`.

**Comment by falconmick** [comment]
Thanks to everyone who put in the hard work!

**Comment by cyio** [comment]
@chris001 `ugrade` => `upgrade`

**Comment by ciprianflorescu** [comment]
Thanks for your answers, I already bought a cert from a provider.

@falconmick So this time if something doesn't work, I can actually get that refund you were so concerned about.

**Comment by HaeckDesign** [comment]
Thank you to the whole Certbot team!!  We're starting a monthly donation to EFF in your honor (https://supporters.eff.org/donate) - If there's a way to earmark it for your team, please let us know.

**Comment by jfmessierdotca** [comment]
Just to mention that the command that was previously shown for a system by "BMW" actually works as-is for Raspbian OS on my Raspberry pi:

`certbot --authenticator standalone --installer apache -d <yourdomain> --pre-hook "systemctl stop apache2" --post-hook "systemctl start apache2"`

I only substituted "-d <yourdomain>" with the multiple ones I have for my cert. And since I have several domain names, I put the whole thing in a script. Works perfectly for me.

**Comment by gurussia** [comment]
guys certbot is SHIT. Today we try implement lets encrypt and we try legally use certbot. Now we read that lets encrypt has wildcard so we try certbot certonly -d  *.yourdomain.com --standalone and it comment error that if it remember that certbot don't support. So our coworker try developer update and he install certbot when it give same command certbot certonly -d *.yourdomain.com --standalone it comment the error upper that other peoples create client with the curr.... THIS IS SHIT. becouse lets ecnrypt COMMENT certbot HAS WORKING ACME V2 SUPPORT THAT *.YOURDOMAIN.COM wildcard work. Now peoples ask yourself why there is no free ssl certificate. Lets ecnrypt do it for 3 months free YES thanks. But if someone give you SHIT, YES HUMAN SHIT free are you gonna EAT IT. Why not, this SHIT is free. SO lets encrypt and certbot ARE THE SAME SHITS. they are human shits. lets encrypt can make it free forever and renew it every years but lets encrypt DON'T WANT IT. All our .com and .cn domains are working not with the THANKS of the lets ecnrypt. We love .cn becouse here in CHINA our .cn are very humankind that has suprise for chinese peoples it is shame that west country ARE LAUGHTING for humankind give them FREE SHITS and they will eat them.

**Comment by TomasHubelbauer** [comment]
@gurussia Running into problems using software is frustrating for sure, even free software, but ultimately everyone in the free software community is driven by contribution to a common goal and I'm sure if you'll take a minute to write down what problems you are facing specifically, people will be more than willing to help. I'm not affiliated with Let's Encrypt, but I am a very, very satisfied user and so it saddens me to see such messages, even though I understand the frustration behind them. Please consider following up with a message that can get us off the ground trying to troubleshoot and resolve your problem and maybe the result will change your opinion of LE and Certbot.

**Comment by bmw** [comment]
On that note, I think this resolved issue has outlived its usefulness. As always, if anyone has an honest issue using Certbot or has a bug report or feature request, please feel free to make a new issue or topic on https://community.letsencrypt.org respectfully explaining it and people will be happy to help you.
